src/Raven.Server/Documents/Handlers/Debugging/DocumentDebugHandler.cs

```diff
@@ -54,5 +57,89 @@ public async Task HugeDocuments()
                 writer.WriteEndObject();
             }
         }
+        
+        [RavenAction("/databases/*/debug/documents/scan-corrupted-ids", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
+        public async Task ScanCorruptedIds()
+        {
+            var startEtag = GetIntValueQueryString("startEtag", required: false) ?? 0;
+            var corruptedCount = GetIntValueQueryString("corruptedCount", required: false) ?? 1024;
+            
+            var corrupted = new List<string>();
+            long? lastEtag = null;
+            
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                await using (var writer = new AsyncBlittableJsonTextWriter(context, Stream.Null))
+                using (context.OpenReadTransaction())
```

Comment:
Can we do the processing in multiple transactions / batches? I'm worried that on huge dbs that can take a while and will affect temporary spaces used by a database since this long running read tx will block clearing the scratch.buffers files

Reply:
Basically, it is a very fast operation, and you can also stop it by aborting the request.  
But I will add logic here to release the read transaction.  
What about doing that due to time?  
Maybe every 1 minute.  
I can also add an option to modify it with a query parameter.

Reply:
The problem is that this _isn't_ fast if you need to read a lot of data from disk. And you may have the tx holds flushing for a long while afterward.